### ISSUE TITLE
layoutScroll breaking layoutRoot in fixed elements

### ISSUE THREAD

**Issue by jamesvclements** [framer/motion#2111]
My site has a nav and island, both position: fixed to the top of the screen:

![localhost3000about-Friday-May-05-2023-04 27 39PM@2x](https://user-images.githubusercontent.com/20052710/236580430-63e1bbb4-8219-4add-97af-47352d5015e4.png)

Both components are part of the root `layout.tsx`, so they're shared across pages.

I can't find a way to get `layoutScroll` and `layoutRoot` to work together, so that:
1. The island stays in place between route changes when you're scrolled down the page and click a nav link
2. The subnav highlight only animates X, and not Y from some offscreen vaue

Here's a video of what happens **without** `layoutScroll`:
https://share.cleanshot.com/47PdNddx

The island bounces all over between page changes.

Here's a video of what happens **with** `layoutScroll`:
https://share.cleanshot.com/fdHQ7xGy

The island stays in place, but the subnav highlight animates from some "y" value that doesn't make sense, instead of only animating the "x" as it does in the first video.

Is it possible to have both? Thanks!

**Comment by mikirejf** [comment]
It would be helpful if you could provide a codesandbox where the issue can be reproduced.

**Comment by jamesvclements** [comment]
@mikirejf yep I'll try, need to reproduce a minimal multi-page next.js with app dir to replicate

**Comment by jamesvclements** [comment]
@mikirejf added a code sandbox and more details!

**Comment by jakobprufer** [comment]
Having the same issue, have you found a solution?

**Comment by ShacharHarshuv** [comment]
I have a similar issues only I didn't do any of the things the post said were the culprit. My "highlight" element doesn't have animation presence or any enter/exit animations or variants. 
What else could be causing this?

**Comment by mattgperry** [comment]
im closing this out as the original example isnt available any more but if someone has a reproduction please open a new ticket and I'll take a look
